Add employee on entity map

lomamech opened the pull request:
- Add the Path of EmployeeStanding Report in config/identifiers
- Update the API for Employee Allow the Possibility to set creditor and
  debitor uuid from Client
- Update referenceLookup track when the code is for Employee for redirect
  to Employee Standing Reports
- Add New Trigger to set employee on entity_map table
- Update Data fore database of test


test(Employee Reference)  …
- Implement E2E test for Research Reference Of Employee
- Update Api for Employee And Filtering of reference of Employee

Closes #2790

lomamech committed:
refactor(Employee in Entity Map)
- Add the Path of EmployeeStanding Report in config/identifiers
- Update the API for Employee Allow the Possibility to set creditor and
  debitor uuid from Client
- Update referenceLookup track when the code is for Employee for redirect
  to Employee Standing Reports
- Add New Trigger to set employee on entity_map table
- Update Data fore database of test

lomamech committed:
test(Employee Reference)
- Implement E2E test for Research Reference Of Employee
- Update Api for Employee And Filtering of reference of Employee

lomamech committed:
Sanitaze code

lomamech committed:
Sanitaze Code

jniles commented on client/src/i18n/en/table.json:
```diff
@@ -23,7 +23,7 @@
             "CHARGE": "Charge",
             "CLIENT": "Client",
-            "CODE": "Code",
+            "CODE": "Registration number",
```
It looks like this is used in other modules (like [stock](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/stock/entry/entry.js#L72)) for the word "Code".  Could you make a new key for this translation?

jniles commented on client/src/modules/employees/registry/registry.js:
```diff
@@ -38,6 +38,12 @@ function EmployeeRegistryController(
+    { field                : 'reference',
+      displayName          : 'TABLE.COLUMNS.REFERENCE',
+      aggregationType      : uiGridConstants.aggregationTypes.count,
+      aggregationHideLabel : true, headerCellFilter     : 'translate',
```
This should be on two lines:
```js
aggregationHideLabel: true,
headerCellFilter: 'translate',
```

jniles commented on server/config/identifiers.js:
```diff
@@ -20,6 +20,7 @@ module.exports = {
   EMPLOYEE : {
     key   : 'EM',
     table : 'employee',
+    documentPath : '/reports/finance/employeeStanding/',
```
Nice! 👍

jniles commented on test/data.sql:
```diff
@@ -615,9 +615,9 @@ INSERT INTO `creditor_group` VALUES
 -- Creditors
 INSERT INTO `creditor` VALUES
-  (HUID('42d3756a-7770-4bb8-a899-7953cd859892'), HUID('b0fa5ed2-04f9-4cb3-92f7-61d6404696e7'), 'Personnel'),
+  (HUID('42d3756a-7770-4bb8-a899-7953cd859892'), HUID('b0fa5ed2-04f9-4cb3-92f7-61d6404696e7'), 'Crediteur[Test 2 Patient]'),
```
Did these need to change...?

lomamech replied:
Just to respect the formatting of the generate text when creating a creditor

jniles replied:
Ah, okay.  Makes sense.

jniles commented on test/end-to-end/employees/registry.spec.js:
```diff
@@ -57,6 +58,14 @@ describe('Employees Registry', () => {
     employeeRegistryPage.clearFilter();
   });
 
+  it(`should find One employee With reference "${parameters.reference}"`, () => {
```
Nice 👍

jniles reviewed (commented):
Looks pretty good to me!

It looks like multiple other modules use `CODE` for translation where the new text wouldn't work.  Could you make a new key?

lomamech committed:
improvement(Employee Data)
- Restaure Code in Translate Key Code
- Add a new key for Registration Number
- Use credit balance or debit balance for Employee Standing and Update
  report Employee Standing

jniles reviewed (approved):
LGTM.  All my remarks were addressed.

bors r+
